src/callTraceStorage.cpp

```diff
@@ -289,3 +289,18 @@ void CallTraceStorage::add(u32 call_trace_id, u64 counter) {
         }
     }
 }
+
+void CallTraceStorage::resetCounters() {
+     for (LongHashTable* table = _current_table; table != NULL; table = table->prev()) {
+        u64* keys = table->keys();
+        CallTraceSample* values = table->values();
+        u32 capacity = table->capacity();
+
+        for (u32 slot = 0; slot < capacity; slot++) {
+            if (keys[slot] != 0) {
+                CallTraceSample& s = values[slot];
+                storeRelease(s.counter, 0);
```

Comment:
Don't we need to reset `s.samples` too?
Check that flame graph is generated correctly both with and without `--total` option.